### ISSUE TITLE
Test Timing out on Github Actions  - Cypress 7.0 

### ISSUE THREAD

**Issue by Josehower** [cypress-io/cypress#15990]
### Current behavior
On Cypress 7.0 Test is failing on GitHub Actions due Timeout, Locally the test is working good
This test never failed before on older cypress versions

## Local Test

![image](https://user-images.githubusercontent.com/61600906/114750703-8f2f7d80-9d54-11eb-9f44-e11328234dea.png)

## Github Actions

![image](https://user-images.githubusercontent.com/61600906/114751876-fe59a180-9d55-11eb-9a20-b83e00558099.png)

### Desired behavior

This test should be working as good as always

### Test code to reproduce

# Test Code

```js
describe('Navigation', () => {
  it('checks that the app reads the cookie as expected and shows the right courses based on its authorization', () => {
    cy.visit('/');
    cy.get('[data-cy=prep-course]').should('not.exist');
    cy.get('[data-cy=btcmp-course]').should('not.exist');
    cy.setCookie('courses', cookiePrepPreview);
    cy.visit('/');
    cy.get('[data-cy=prep-course]').should('be.visible');
    cy.get('[data-cy=btcmp-course]').should('not.exist');
    cy.get('preview-lock').should('be.visible');
    cy.setCookie('courses', cookiePrep);
    cy.visit('/');
    cy.get('[data-cy=prep-course]').should('be.visible');
    cy.get('[data-cy=btcmp-course]').should('not.exist');
    cy.setCookie('courses', cookieBoth);
    cy.visit('/');
    cy.get('[data-cy=btcmp-course]').should('be.visible');
    cy.get('[data-cy=prep-course]').should('be.visible');
  });

  it('checks that the links for the courses work as expected and the courses pages show the right content', () => {
    cy.setCookie('courses', cookieBoth);
    cy.visit('/');
    cy.get('[data-cy=prep-course]').should('be.visible').click();
    cy.get('[data-cy=bootcamp-prep]').should('be.visible');
    cy.url().should('include', 'prep-l-webfs-gen-0');
    cy.contains('001 - HTML: Definition');
    cy.contains('002 - HTML: Elements, Tags & Attributes');
    cy.contains('003 - HTML: Interactive Elements');
    cy.contains('004 - CSS: Definition');
    cy.contains('005 - CSS: Selectors & Declarations');
    cy.contains('006 - CSS: Selectors in Detail');
    cy.contains('007 - CSS: Layout');
    cy.contains('008 - JS: Definition');
    cy.contains('009 - JS: Variables');
    cy.contains('010 - JS: Functions');
    cy.visit('/');
    cy.get('[data-cy=btcmp-course]').should('be.visible').click();
    cy.url().should('include', 'btcmp-l-webfs-gen-0');
    cy.contains('001 - Tasks');
    cy.contains('002 - Documentation');
    cy.contains('Cheatsheet');
  });
});

```
### Versions

Last working version

    `"cypress": "6.8.0"`
    
Unexpected behavior version

    `"cypress": "7.0.1"`

### additional comments

I can confirm that the test duration is significantly longer than the previous version

![image](https://user-images.githubusercontent.com/61600906/114752126-424ca680-9d56-11eb-97a4-7351486dea86.png)

**Comment by fourcolors** [comment]
I have a very similar issue right now that's been affecting several tests at our company. Locally it's working, even using the electron browser but on circle ci it fails. 

![image](https://user-images.githubusercontent.com/51751/114781123-7cf21700-9d35-11eb-9245-3845754a1e0f.png)

```
  cy.get('[aria-expanded=true]').should('exist');
```
We are running `"cypress": "^6.8.0",`

All I can think of is that on CI it's headless. I'm not sure why this would make selectors randomly fail though. 

We found this 👇🏻 helps and seperating out the test but I believe this should be working
```
cy.get('[data-cy=selector]').then(($el) => {
  Cypress.dom.isVisible($el) // true
})
```

**Comment by bahmutov** [comment]
I think the longer timing due to performance regression is affecting the tests. For now I suggest sticking with the last known Cypress version or trying increasing the default command timeout setting in the cypress.json file.

**Comment by Hypercubed** [comment]
I was attempting to repoduce a issue we were seeing in our code base here https://github.com/Hypercubed/cypress-issue that looks related this ticket.  I am hesitant to post this as a bug since I think, for the most part, it's a flaky test on our end.  However, it does show that there is some sort of performance regression in v7.x vs v6.  Increasing command timeout doesn't help on our case since it boils down to how long takes takes cypress to run the next command when other things are going on in the app.  There has been nothing I can find that can force Cypress execute the commands fast enough to capture the DOM.  Hope this info helps.

**Comment by OlgaShchekina** [comment]
We had the same issue. I downgraded Cypress to 6.8.0. Will this be fixed? Now I cannot use the new Cypress features that were added in the latest versions.

**Comment by jennifer-shehane** [comment]
Can you all try updating to Cypress 8.0 and report if this issue is still happening? We've addressed some performance issues that were introduced with the screen size changes.

**Comment by OlgaShchekina** [comment]
It seems like it's working now. We will test for a couple more days. 

But the cy.stub() does not work and neither do the recipes from here: https://glebbahmutov.com/blog/stub-window-open/  It worked well with the older version.

<img width="672" alt="Screen Shot 2021-07-20 at 4 52 10 PM" src="https://user-images.githubusercontent.com/57824176/126409508-7041ab40-05ca-4623-8074-99a459d17cc1.png">

**Comment by jennifer-shehane** [comment]
Closing as I believe this is resolved in 8.0. 

In 7.0 the screen size was increased when running headlessly. This will slow down runtime and in some cases with low resourced machines (as GitHub Actions may be depending on the background process) it can cause hanging. In 8.0 we reverted this change back to pre-7.0 screen size.
